service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/GetActionManagedIdentityStep.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.managedIdentity;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gets an Azure Managed Identity for a user's 'action managed identity', which may be used to grant
+ * users access to private ACRs. This step will insert a fetched action identity into the flight
+ * context if one can be obtained from Sam.
+ */
+public class GetActionManagedIdentityStep implements Step {
+
+  public static final String ACTION_IDENTITY = "ACTION_IDENTITY";
+  private final SamService samService;
+  private final WorkspaceService workspaceService;
+  private final AuthenticatedUserRequest userRequest;
+  private final UUID workspaceId;
+  private final Logger logger = LoggerFactory.getLogger(GetActionManagedIdentityStep.class);
+
+  public GetActionManagedIdentityStep(
+      SamService samService,
+      WorkspaceService workspaceService,
+      UUID workspaceId,
+      AuthenticatedUserRequest userRequest) {
+    this.samService = samService;
+    this.workspaceService = workspaceService;
+    this.workspaceId = workspaceId;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    String billingProfileId = workspaceService.getWorkspace(workspaceId).spendProfileId().getId();
+    logger.info(
+        "Querying Sam for action identity using billing profile id: '{}'", billingProfileId);
+
+    Optional<String> actionIdentity =
+        samService.getActionIdentityForUser(billingProfileId, userRequest);
+    if (actionIdentity.isPresent()) {
+      logger.info("Fetched action managed identity '{}' from sam.", actionIdentity.get());
```

Comment:
Similar points about including workspace ID in the logging statements in this step for debug-ability.